From 991cfb9bfe0d1e1519f56b5371950e0c20fd1f09 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:24 -0400
Subject: [fs] btrfs: make FS_INFO ioctl available to anyone

Message-id: <1407179109-13714-318-git-send-email-sandeen@redhat.com>
Patchwork-id: 87227
O-Subject: [RHEL7.1 PATCH 317/422] btrfs: make FS_INFO ioctl available to anyone
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit e4ef90ff6198ac58d4c7dbbeae8cc7b24867d1ca
Author: David Sterba <dsterba@suse.cz>
Date:   Thu Apr 24 15:09:18 2014 +0200

    btrfs: make FS_INFO ioctl available to anyone

    This ioctl provides basic info about the filesystem that can be obtained
    in other ways (eg. sysfs), there's no reason to restrict it to
    CAP_SYSADMIN.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 92a5184..2db12fc 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -2557,9 +2557,6 @@ static long btrfs_ioctl_fs_info(struct btrfs_root *root, void __user *arg)
  struct btrfs_fs_devices *fs_devices = root->fs_info->fs_devices;
  int ret = 0;
 
- if (!capable(CAP_SYS_ADMIN))
-  return -EPERM;
-
  fi_args = kzalloc(sizeof(*fi_args), GFP_KERNEL);
  if (!fi_args)
   return -ENOMEM;
-- 
1.7.1